content/cookbook/programs/programs.md

```diff
@@ -0,0 +1,420 @@
+
+@[code](@/code/programs/create-pda/program/src/lib.preview.rs)
+
+  </template>
+
+  </SolanaCodeGroupItem>
+</SolanaCodeGroup>
+
+One can send the required accounts via client as follows
+
+<SolanaCodeGroup>
+  <SolanaCodeGroupItem title="TS" active>
+
+  <template v-slot:default>
+
+@[code](@/code/programs/create-pda/client/main.en.ts)
+
+  </template>
+
+  <template v-slot:preview>
+
+@[code](@/code/programs/create-pda/client/main.preview.en.ts)
+
+  </template>
+
+  </SolanaCodeGroupItem>
+</SolanaCodeGroup>
+
+## How to read accounts
+
+Almost all instructions in Solana would require atleast 2 - 3 accounts, and they
+would be mentioned over the instruction handlers on what order it's expecting
+those set of accounts. It's fairly simple if we take advantage of the `iter()`
+method in Rust, instead of manually indicing the accounts. The
+`next_account_info` method basically slices the first index of the iterable and
+returning the account present inside the accounts array. Let's see a simple
+instruction which expects a bunch of accounts and requiring to parse each of
+them.
+
+<SolanaCodeGroup>
+  <SolanaCodeGroupItem title="Rust" active>
+
+  <template v-slot:default>
+
+@[code](@/code/programs/read-account/program/src/lib.rs)
+
+  </template>
+
+  <template v-slot:preview>
+
+@[code](@/code/programs/read-account/program/src/lib.preview.rs)
+
+  </template>
+
+  </SolanaCodeGroupItem>
+</SolanaCodeGroup>
+
+## How to verify accounts
+
+Since programs in Solana are stateless, we as a program creator have to make
+sure the accounts passed are validated as much as possible to avoid any
+malicious account entry. The basic checks one can do are
+
+1. Check if the expected signer account has actually signed
+2. Check if the expected state account's have been checked as writable
+3. Check if the expected state account's owner is the called program id
+4. If initializing the state for the first time, check if the account's already
+   been initialized or not.
+5. Check if any cross program ids passed (whenever needed) are as expected.
+
+A basic instruction which initializes a hero state account, but with the above
+mentioned checks is defined below
+
+<SolanaCodeGroup>
+  <SolanaCodeGroupItem title="Rust" active>
+
+  <template v-slot:default>
+
+@[code](@/code/programs/verify-account/program/src/lib.rs)
+
+  </template>
+
+  <template v-slot:preview>
+
+@[code](@/code/programs/verify-account/program/src/lib.preview.rs)
+
+  </template>
+
+  </SolanaCodeGroupItem>
+</SolanaCodeGroup>
+
+## How to read multiple instructions from a transaction
+
+Solana allows us to take a peek at all of the instructions in the current
+transaction. We can store them in a variable and iterate over them. We can do
+many things with this, like checking for suspicious transactions.
+
+<SolanaCodeGroup>
+  <SolanaCodeGroupItem title="Rust" active>
+
+  <template v-slot:default>
+
+@[code](@/code/programs/read-multiple-instructions/program/lib.rs)
```

Comment:
Just checking do these files already exist in the repo? I would have thought they'd be in this PR.